### PR TITLE
Refactor drinkers to have one object array

### PR DESCRIPTION
**Pull request by nzpete**

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -160,9 +159,18 @@ function addUserFromMessage(message) {
...
+		var duplicate_hand = false;
+		for (var i = 0; i < state.drinkers.length; i++) {
+			if (state.drinkers[i].name == name) {
+				duplicate_hand = true;
+				break;
+			}
+		} 
+
+		if (duplicate_hand) {
 			bot.reply(message, fromArray([
```
Comment (DanHulton) [inline bot.js]: Can move this logic inside the if() check above and get rid of the duplicate_hand variable entirely.

File: bot.js [hunk 2]
```diff
@@ -183,8 +191,11 @@ function addUserFromMessage(message) {
 
 		// Add user to list
 		state.available--;
-		state.drinkers_name.push(name);
-		state.drinkers_handle.push(handle);
+		state.drinkers.push({
+			"name": name,
```
Comment (DanHulton) [inline bot.js]: I'm seeing wonky formatting here, but it's probably just a spaces/tabs issue and github being silly.

File: bot.js [hunk 3]
```diff
@@ -183,8 +191,11 @@ function addUserFromMessage(message) {
 
 		// Add user to list
 		state.available--;
-		state.drinkers_name.push(name);
-		state.drinkers_handle.push(handle);
+		state.drinkers.push({
```
Comment (DanHulton) [inline bot.js]: A neat feature of modern JS is that if your object property and value are the same (like "name": name"), you can just write it once:

`state.drinkers.push({ name, id, handle });`

Though you'll have to change your variable declaration above from `var user_id = response.user.id;` to `var id = response.user.id;`.

File: bot.js [hunk 4]
```diff
@@ -203,24 +214,23 @@ function addUserFromMessage(message) {
 function finish(message) {
 	bot.reply(message, "That's it!  Here's who gets coffee:");
 
-	for (var i = 0; i < state.drinkers_name.length; i++) {
-		bot.reply(message, "- " + state.drinkers_name[i]);
+	for (var i = 0; i < state.drinkers.length; i++) {
```
Comment (DanHulton) [inline bot.js]: Node.js has a recent-enough version of ecmascript that you can rewrite this into the more-understandable:

```for (let drinker of state.drinkers) {
    bot.reply(message, "- " + drinker.name);
}

Comment (DanHulton) [inline bot.js]: Node.js has a recent-enough version of ecmascript that you can rewrite this into the more-understandable:

```for (let drinker of state.drinkers) {
    bot.reply(message, "- " + drinker.name);
}

File: bot.js (hunk 2 above)
Comment (nzpete) [inline bot.js]: Thanks... I'll come chat about how that should be formatted.

File: bot.js [hunk 5]
```diff
@@ -203,24 +214,23 @@ function addUserFromMessage(message) {
...
+		bot.reply(message, "- " + state.drinkers[i].name);
 	}
 }
 
 function ready(message) {
 	bot.reply(message, "Come get your coffee:");
 
-	for (var i = 0; i < state.drinkers_handle.length; i++) {
-		bot.reply(message, "- @" + state.drinkers_handle[i]);
+	for (var i = 0; i < state.drinkers.length; i++) {
```
Comment (DanHulton) [inline bot.js]: Can rewrite as `for ... of` here as well.

File: bot.js (hunk 1 above)
Comment (nzpete) [inline bot.js]: I am confused here.  I will come chat.

**Comment by nzpete** [comment]
@DanHulton Thanks for the advice.  I have addressed the comments.  What do you think.

File: bot.js [hunk 6]
```diff
@@ -203,24 +207,23 @@ function addUserFromMessage(message) {
...
 	}
 }
 
 function ready(message) {
 	bot.reply(message, "Come get your coffee:");
 
-	for (var i = 0; i < state.drinkers_handle.length; i++) {
-		bot.reply(message, "- @" + state.drinkers_handle[i]);
+	for (let drinker of state.drinkers) {
+		bot.reply(message, ` - @${drinker.handle} `);
```
Comment (DanHulton) [inline bot.js]: ``bot.reply(message, `- <@${drinker.id}|${drinker.name}>`);``

**Comment by DanHulton** [comment]
:+1:

File: bot.js (hunk 6 above)
Comment (nzpete) [inline bot.js]: Looks good except should be drinker.handle (not drinker.name)
Fixed.

**Pull request merged**